### ISSUE TITLE
Incorrect case in Cloudflare Logpush pipeline

### ISSUE THREAD

**Issue by tiaanwest** [elastic/integrations#8737]
In the definition for the pipelines, for instance https://github.com/elastic/integrations/blob/main/packages/cloudflare_logpush/data_stream/audit/elasticsearch/ingest_pipeline/default.yml#L28, we use `ctx.json?.When` when it should be `ctx.json?.when`, as per the Cloudflare API definition: https://developers.cloudflare.com/api/operations/audit-logs-get-account-audit-logs

The current definitions throws the following error when ingesting logs:

```
"error.message": [
"Processor \"conditional\" in pipeline \"logs-cloudflare_logpush.audit-1.17.2\" failed with message \"Cannot invoke \\\"Object.getClass()\\\" because \\\"value\\\" is null\""
],
```

Using `.when` (lowercase) instead of `.When` (uppercase) it works as expected.

**Comment by jamiehynds** [comment]
Thanks for reporting @tiaanwest. Would you mind having a look @chemamartinez?

**Comment by chemamartinez** [comment]
Hi @tiaanwest,

Did you find that error in a real scenario with real logs? If so, that should be caused by a field change in the Cloudflare's side and it means not only `when` but all the fields would be lowercase now.

I could not find any current audit log, but we have some samples from our last development a few months ago and every field was uppercase. In addition, they are still documented that way in their developers guide: https://developers.cloudflare.com/logs/reference/log-fields/account/audit_logs/

**Comment by gils3nan** [comment]
We are seeing this error now when using the `cloudflare_logpush` integration.
Noting the cloudflare documentation above with the `audit_logs` fields with the `New` being uppercase. The ingest pipeline does include `New` in the relevant conditions on some processors and it also uses `New` in the relevant script processor.

We are still seeing the below error, from the `logs-cloudflare_logpush.audit-1.21.0` ingest pipeline:
```
"error": {
      "message": [
        "Processor \"conditional\" in pipeline \"logs-cloudflare_logpush.audit-1.21.0\" failed with message \"Cannot invoke \\\"Object.getClass()\\\" because \\\"value\\\" is null\""
      ]
    }
```

We have also seen other errors from the below pipelines (Though these seem to be simple `Set` processor changes, we have yet to test this out/review this ourselves):

Pipeline: `logs-cloudflare_logpush.gateway_dns-1.21.0`
Error Message:
```
"error": {
      "message": [
        "Processor \"set\" in pipeline \"logs-cloudflare_logpush.gateway_dns-1.21.0\" failed with message \"field [host] not present as part of path [cloudflare_logpush.gateway_dns.host.id]\""
      ]
    }
```

Pipeline: `logs-cloudflare_logpush.gateway_http-1.21.0`
Error Message:
```
"error": {
      "message": [
        "Processor \"set\" in pipeline \"logs-cloudflare_logpush.gateway_http-1.21.0\" failed with message \"field [action] not present as part of path [cloudflare_logpush.gateway_http.action]\""
      ]
    }
```

**Comment by gils3nan** [comment]
Hello, 

Has there been any update on this as we are still seeing this in version `1.24.0`. Showing the same messages as above.

Pipelines in question are:

**logs-cloudflare_logpush.audit-1.24.0** 
Showing Error:
```
Processor "conditional" in pipeline "logs-cloudflare_logpush.audit-1.24.0" failed with message "Cannot invoke \"Object.getClass()\" because \"value\" is null"
```

**logs-cloudflare_logpush.gateway_dns-1.24.0**
Showing Error:
```
Processor "set" in pipeline "logs-cloudflare_logpush.gateway_dns-1.24.0" failed with message "field [host] not present as part of path [cloudflare_logpush.gateway_dns.host.id]"
```

**logs-cloudflare_logpush.gateway_http-1.24.0**
Showing Error:
```
Processor "set" in pipeline "logs-cloudflare_logpush.gateway_http-1.24.0" failed with message "field [action] not present as part of path [cloudflare_logpush.gateway_http.action]"
```

**Comment by gils3nan** [comment]
Hey @chemamartinez - thank you for help with this.

Please see some examples:

- logs-cloudflare_logpush.audit-1.24.0
Example of `event.original`
```
{
    "ActorIP": "",
    "ID": "23e9309a-9851-4a45-a759-2b7c84bef7da",
    "ResourceType": "image",
    "When": 1726572009660000000
}
```


- logs-cloudflare_logpush.gateway_dns-1.24.0
Example of `event.original`
```
{
    "Datetime": 1726572053335000000
}
```


- logs-cloudflare_logpush.gateway_http-1.24.0
Example of `event.original`
```
{
    "Datetime": 1726572053335000000
}
```

The events seen in `logs-cloudflare_logpush.gateway_dns-1.24.0` and `logs-cloudflare_logpush.gateway_http-1.24.0` are very similar. With the error messages showing a similar error message when parsing around the fields not being present. Which ofc, would be able to be worked around in the ingest pipeline to allow (ignore on processor failure) to more forward in the ingest pipeline. But this would change on upgrade of the integration.

Also note, that the actual values in the Cloudflare `event.original` fields, the data is lacking.

**Comment by gils3nan** [comment]
I fear, that a review of this Cloudflare configuration from our side may be needed? As from the event.original values/data. It does seem like there is minimal to no relevant or useful data being seen?

**Comment by chemamartinez** [comment]
> I fear, that a review of this Cloudflare configuration from our side may be needed? As from the event.original values/data. It does seem like there is minimal to no relevant or useful data being seen?

Yeah, that `event.original` field should contain the whole event, the first step in the ingest pipeline is to copy the whole incoming log into it.

Regarding the configuration, you could verify that you have enabled the setting `Preserve original event` but it should be, otherwise you wouldn't see any `event.original` field at all.

On the other hand, maybe you can identify that exact same event at the source (don't know if you are pulling data from a S3 bucket or directly from Cloudflare), but could be interesting to see if they appear complete in the source or they are just malformed from the beginning.

Thanks!

**Comment by gils3nan** [comment]
Hey @chemamartinez - thank you again for your help with this!

We can confirm the parsing errors we have been seeing was due to a mis-configuration in Cloudflare. The log settings weren't set up correctly it seemed.

We are seeing all the logs in Elastic and see that they are all being parsed correctly now. We still have the `event.original` enabled still and can see the data is actually there now!

**Comment by chemamartinez** [comment]
Glad to hear that @gils3nan!

@tiaanwest should we keep this issue still open? I couldn't find any log sample where the field you mentioned is lowercase. Could you provide some?
